src/domain/challenge/hub/hub.service.ts

```diff
@@ -256,7 +278,14 @@ export class HubService {
       .leftJoinAndSelect('challenge.opportunities', 'opportunities')
       .getMany();
 
-    const sortedHubs = hubsDataForSorting.sort((a, b) => {
+    const visibleHubs = hubsDataForSorting.filter(hub => {
```

Comment:
You could do the filter and sort in the query itself. It will be much faster and the code will be less.

Reply:
Yes of course that is faster. But I also wanted to go through the same code path (in the hubFilter service) and also not spend time working a more complex db query. If in the future the number of hubs gets into the 10,000 range then worth considering optimising this, but could not justify that now.